tidb-binlog/driver/reader/offset_test.go

```diff
@@ -20,25 +20,24 @@ import (
 	"time"
 
 	"github.com/Shopify/sarama"
-	. "github.com/pingcap/check"
 	pb "github.com/pingcap/tidb/tidb-binlog/proto/go-binlog"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestClient(t *testing.T) {
-	TestingT(t)
-}
-
-var _ = Suite(&testOffsetSuite{})
-
 type testOffsetSuite struct {
+	suite.Suite
 	producer  sarama.SyncProducer
 	config    *sarama.Config
 	addr      string
 	available bool
 	topic     string
 }
 
-func (to *testOffsetSuite) SetUpSuite(c *C) {
+func TestClient(t *testing.T) {
+	suite.Run(t, new(testOffsetSuite))
```

Comment:
Please not use the ```suite```. it will cause the data race in #32747.

Reply:
Well, I will revise it, mainly because I see that suite is also written in the tidb project.

Reply:
This is ok with only one test, if there are multiple tests, should I call the setup function before each test?

Reply:
> Well, I will revise it, mainly because I see that suite is also written in the tidb project.

the other test case have been in the plan that they will be rewritten.

Reply:
> This is ok with only one test, if there are multiple tests, should I call the setup function before each test?

Yes, You can.

Reply:
done